### ISSUE TITLE
Support Go Modules not properly integrated

### ISSUE THREAD

**Issue by farrukhny** [dimfeld/httptreemux#66]
I think integration of go modules is good but i think it should not include version/release in path

```module github.com/dimfeld/httptreemux/v5```

Issues I have with that is all my code import path was based on   github.com/dimfeld/httptreemux and i can used to just call httptreemux.NAME OF FUNC now i have to change all my import path for httptreemux by adding alias like 

import httptreemux "github.com/dimfeld/httptreemux/v5"

also in case of creating new version/release you have to change path ?

please take a looks this example: https://github.com/lib/pq

**Comment by SamWhited** [comment]
> I think it should not include version/release in path

If you don't do this for modules with a major version >1, module support will be broken.

> Issues I have with that is all my code import path was based on github.com/dimfeld/httptreemux and i can used to just call httptreemux.NAME OF FUNC now i have to change all my import path for httptreemux by adding alias like

You can still do this; the /v5 is a virtual component of the path, the package name is still httptreemux, so that import you're using isn't actually doing anything or renaming anything unless you're on an old version of Go that doesn't understand modules.

All versions of Go from 1.9.7+, 1.10.3+, and higher major versions understand the /v5 in the import path, so if you're using a version of Go that's supported and receiving security updates (1.10 or 1.11) you should be fine.

> also in case of creating new version/release you have to change path ?

Only on major version releases that break backwards compatibility, in which case having them have two different import lines means you can include both temporarily while you upgrade (eg. you can import /v5 in one place and /v6 in another).

> please take a looks this example: https://github.com/lib/pq

That is still below version 2; you don't include the major version in the path for v1 modules.

**Comment by dimfeld** [comment]
Thanks, Sam, for the detailed explanation. Unfortunately I can't reset to version 1 since that would break things for everyone else using an older version of the package with some other dependency manager. And even if I did, if I had to make a breaking change in the future, then the version suffix would come back again.

This is just how Go modules works, so I'll close this issue. On the bright side, this package is pretty much in maintenance mode, so breaking changes are pretty rare.
